plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-footer.tsx

```diff
@@ -2,21 +2,84 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { createInterpolateElement } from '@wordpress/element';
+import {
+	createInterpolateElement,
+	useCallback,
+	useContext,
+	useEffect,
+} from '@wordpress/element';
 import { Link } from '@woocommerce/components';
+import { Spinner } from '@wordpress/components';
+// @ts-expect-error Missing type in core-data.
+import { __experimentalBlockPatternsList as BlockPatternList } from '@wordpress/block-editor';
+
 /**
  * Internal dependencies
  */
 import { SidebarNavigationScreen } from './sidebar-navigation-screen';
 import { ADMIN_URL } from '~/utils/admin-settings';
+import { useEditorBlocks } from '../hooks/use-editor-blocks';
+import { usePatternsByCategory } from '../hooks/use-pattern';
+import { HighlightedBlockContext } from '../context/highlighted-block-context';
+
+const footerPatternNames = [
+	'woocommerce-blocks/footer-large',
+	'woocommerce-blocks/footer-large-dark',
+	'woocommerce-blocks/footer-simple',
+	'woocommerce-blocks/footer-simple-dark',
+	'woocommerce-blocks/footer-simple-menu-and-cart',
+	'woocommerce-blocks/footer-with-2-menus',
+	'woocommerce-blocks/footer-with-2-menus-dark',
+	'woocommerce-blocks/footer-with-3-menus',
+];
 
 export const SidebarNavigationScreenFooter = () => {
+	const { isLoading, patterns } = usePatternsByCategory( 'woo-commerce' );
+	const footerPatterns = patterns.filter( ( pattern ) =>
+		footerPatternNames.includes( pattern.name )
+	);
+
+	const { setHighlightedBlockIndex, resetHighlightedBlockIndex } = useContext(
+		HighlightedBlockContext
+	);
+
+	useEffect( () => {
+		setHighlightedBlockIndex( 0 );
+	}, [ setHighlightedBlockIndex ] );
+
+	const [ blocks, onChange ] = useEditorBlocks();
+	const onClickFooterPattern = useCallback(
+		( _pattern, selectedBlocks ) => {
+			const newFooterBlock = {
+				...selectedBlocks[ 0 ],
+				attributes: {
+					...selectedBlocks[ 0 ].attributes,
+					slug: 'footer',
+				},
+			};
+
+			onChange(
+				[
+					...blocks.map( ( block ) => {
+						if ( block.attributes?.slug === 'footer' ) {
```

Comment:
There is an issue (p1694583571487789-slack-C01SFMVEYAK) when using slug. We could replace the last block here. 😃

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/40136/commits/734032701872d77743de45a07add712e3a79fede